lms/djangoapps/support/views/enrollments.py

```diff
@@ -84,6 +84,55 @@ def get(self, request, username_or_email):
 
     @method_decorator(require_support_permission)
     def post(self, request, username_or_email):
+        """Allows support staff to create a user's enrollment."""
+        try:
+            course_id = request.data['course_id']
+            course_key = CourseKey.from_string(course_id)
+            mode = request.data['mode']
+            enrollment_modes = [
+                enrollment_mode['slug']
+                for enrollment_mode in self.get_course_modes(course_key)
+            ]
+            if mode not in enrollment_modes:
+                return HttpResponseBadRequest(
+                    f'{str(mode)} is not a valid mode for {str(course_id)}. '
+                    f'Possible valid modes are {str(enrollment_modes)}'
+                )
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_key)
+            if enrollment is not None:
+                return HttpResponseBadRequest(
+                    f'The user {str(username_or_email)} is already enrolled in {str(course_id)}.'
+                )
+            reason = request.data['reason']
+            enrollment = CourseEnrollment.enroll(
+                user=user, course_key=course_key, mode=mode
+            )
+        except KeyError as err:
```

Comment:
This may seem like a large block but nearly every possible exception is handled by by hook or by crook. Both `get_enrollment` and `enroll` class methods handle their respective exceptions as well.

I think we can keep this.

Reply:
I don't agree. Having small try catch with a specific purpose is better. This also makes the code readability better because one can easily understand the conditions in which the execution would fail.